lib/SILOptimizer/Mandatory/Differentiation.cpp

```diff
@@ -2579,6 +2834,9 @@ class VJPEmitter final
   /// The VJP function.
   SILFunction *vjp;
 
+  /// The adjoint function.
+  SILFunction *adjoint;
+
   /// The primal info.
   std::unique_ptr<PrimalInfo> primalInfo = nullptr;
```

Comment:
`PrimalInfo` should not be a `unique_ptr`. It should be a locally stored value in `VJPEmitter`.

Reply:
That makes sense.
Should `AdjointEmitter` store a reference to parent `VJPEmitter`, so that it can access `VJPEmitter::primalInfo`? I don't believe it does, currently.

Reply:
Yes.